### PR TITLE
Xmc4xxx die temperature sensor

### PR DESCRIPTION
**Pull request by talih0**
Add die temperature sensor for Infineon xmc4xxx.

### CODE REVIEW

File: drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	ret = XMC_SCU_StartTemperatureMeasurement();
+	if (ret != 0) {
+		return -EBUSY;
+	}
+
+	while (XMC_SCU_IsTemperatureSensorBusy()) {
```
Comment (MaureenHelm) [inline drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c:38]: How long does this take? Perhaps insert a sleep to allow other threads to do work?

File: drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	};
+
+	val = XMC_SCU_GetTemperatureMeasurement();
+	data->temp_out = (val - 605) / 2.05f;
```
Comment (MaureenHelm) [inline drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c:42]: Please document where these constants come from

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c (hunk 1 above)
Comment (talih0) [reply drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c:38]: The documentation doesn't actually say. But, I put a counter in the while loop and it reaches ~350 per temperature reading. Also, k_uptime_get() rarely ticks over.  So, I don't think we need to yield here.

File: drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c (hunk 2 above)
Comment (talih0) [reply drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c:43]: Thanks. Yes, updated with a reference to the manual.

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**